source/common/stats/symbol_table_impl.h

```diff
@@ -26,9 +28,21 @@ namespace Stats {
 /** A Symbol represents a string-token with a small index. */
 using Symbol = uint32_t;
 
-/** Efficient byte-encoded storage an array of tokens, which are typically < 127 */
+/**
+ * Efficient byte-encoded storage of an array of tokens. The most common tokens
+ * are typically < 127, and are represented directly. tokens >= 128 spill into
+ * the next byte, allowing for tokens of arbitrary numeric value to be stored.
+ * As long as the most common tokens are low-valued, the representation is
+ * space-efficient. This scheme is similar to UTF-8.
+ */
 using SymbolStorage = uint8_t[];
 
+/**
+ * We encode the byte-size of a StatName as its first two bytes.
+ */
+constexpr uint64_t StatNameSizeEncodingBytes = 2;
+constexpr uint64_t StatNameMaxSize = 65536;
```

Comment:
I'm OK with this being hardcoded, but it would be good to explicitly write the logic of why it's 2^16. (Is this temporary / can this never be changed?)

Reply:
Redefined based on StatNameSizeEncodingBytes, butting the 65536 in comments.